### DISCUSSION TITLE
Can I add Custom Properties To Work flow Definition To Be Set Using The Designer?

### DISCUSSION THREAD

**Discussion by hassanelsherbiny** [elsa-workflows/elsa-core#2541] [Q&A]
I want to add Some Custom Properties To Workflow Definition which i can set it's values throw the designer
for example i want to add checkbox to flag this workflow for specific criteria to be used later which execution 
Like This
![image](https://user-images.githubusercontent.com/20015837/142383199-16a50341-00ce-4faa-afea-3b8d6ce0118d.png)

**Comment by sfmskywalker** [comment]
There's a new feature coming with Elsa 2.4 that allows you to define custom properties for workflow definitions. It's not documented yet however.

**Reply by hassanelsherbiny** [reply]
and when will V2.4 be ready?

**Reply by sfmskywalker** [reply]
Once the [2.4 milestone](https://github.com/elsa-workflows/elsa-core/milestone/13) is clear of issues, I'll give it a week for people to try the [2.4 prerelease packages](https://www.myget.org/gallery/elsa-2). If no new issues entered the milestone, it will be released.

**Comment by dwevedivaibhav** [comment]
Hi @sfmskywalker ,

Can you please give me the idea , how i can add custom properties in work flow definition ? I am unable to see in documentation as i am using elsa 2.8 . 
Please advise.

Thanks

**Reply by sfmskywalker** [reply]
Hi @dwevedivaibhav , although the `WorkflowDefinition` model supports a `CustomAttributes` dictionary that allows you to store application-specific values, this is not exposed to the designer. Also, the designer does not yet expose the proper extensibility points to let you add custom UI (like we can do with activity settings).

If you want to work on that, I'm happy to provide assistance where I can.

**Reply by dwevedivaibhav** [reply]
Oki Sure, let me check that
